tap_github/scraping.py

```diff
@@ -85,3 +87,67 @@ def _scrape_dependents(url: str, logger: logging.Logger) -> Iterable[Dict[str, A
             time.sleep(1)
         else:
             url = ""
+
+
+def parse_counter(
+    tag: Union[Tag, NavigableString, None], logger: logging.Logger
+) -> int:
+    if not tag:
+        return 0
+    try:
+        title = tag["title"]  # type: ignore
+        if isinstance(title, str):
+            title_string = cast(str, title)
+        else:
+            title_string = cast(str, title[0])
+        return int(title_string.strip())
+    except KeyError:
+        logger.debug(
+            f"Could not parse counter {tag}. Maybe the GitHub page format has changed?"
+        )
+        return 0
+
+
+def scrape_metrics(
+    response: requests.Response, logger: Optional[logging.Logger] = None
+) -> Iterable[Dict[str, Any]]:
+    import re
+
+    from bs4 import BeautifulSoup
```

Comment:
any reason why these imports are in the function rather than at file level? `bs4` is already imported at the top anyway, and `re` is a standard module, so it shouldn't cause import problems.

Reply:
bs4 is not imported completely. This is how it was done before but I don't really mind, I agree that re can be moved to the top though

Reply:
I tried to `git blame` it to figure out why we did it that way, but no obvious reason. Let's leave it like this, hopefully it won't have too much of an impact.